src/main/kotlin/sliv/tool/model/project/Project.kt

```diff
@@ -0,0 +1,17 @@
+package sliv.tool.model.project
+
+import java.nio.file.Path
+
+class Project {
+    var currentDirectory: Path? = null
+        set(value) {
+            TODO("Validate")
+            updateFrames()
+        }
+
+    var frames: List<ProjectFrame>? = null
```

Comment:
same, val

Reply:
Project is a mutable object, depending on currently chosen directory list of project frames changes